File: src/test/java/GetSasToken.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;

import static java.net.URLEncoder.encode;
import static java.nio.charset.StandardCharsets.UTF_8;

public class GetSasToken {

    public static void main(String[] args) throws Exception {

        String sas = new GetSasToken().getSasToken("sb://bulk-scan-servicebus-sprod.servicebus.windows.net/",
            "SendSharedAccessKey",
            "OY5LlaqYyddj46W8gHD8QFcKmtE/EDS3sC6Fk0A5S+M=");
        System.out.println(sas);
    }

    private String getSasToken(String resourceUri, String keyName, String key) throws Exception {
        long epoch = System.currentTimeMillis() / 1000L;
        int week = 60 * 60 * 24 * 7;
        String expiry = Long.toString(epoch + week);

        String stringToSign = encode(resourceUri, "UTF-8") + "\n" + expiry;
        String signature = getHmac256(key, stringToSign);
        return "SharedAccessSignature sr=" + encode(resourceUri, "UTF-8")
            + "&sig=" + encode(signature, "UTF-8")
            + "&se=" + expiry
            + "&skn=" + keyName;
    }


    private String getHmac256(String key, String input) throws Exception {
        Mac sha = Mac.getInstance("HmacSHA256");
        SecretKeySpec secretKey = new SecretKeySpec(key.getBytes(), "HmacSHA256");
        sha.init(secretKey);
        Base64.Encoder encoder = Base64.getEncoder();

        return new String(encoder.encode(sha.doFinal(input.getBytes(UTF_8))));
    }
}

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/GetSasToken.kt
package uk.gov.hmcts.reform.bulkscan.orchestrator

import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.test.context.ActiveProfiles
import java.net.URLEncoder
import java.util.Base64
import javax.crypto.Mac
import javax.crypto.spec.SecretKeySpec

private val LOG = LoggerFactory.getLogger(GetSasToken::class.java)

//@SpringBootApplication
@ActiveProfiles("nosb")
class GetSasToken {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            println(GetSasToken().getSasToken(args[0], args[1], args[2]))
        }
    }

    fun getSasToken(resourceUri: String, keyName: String, key: String): String {
        val epoch = System.currentTimeMillis() / 1000L
        val week = 60 * 60 * 24 * 7
        val expiry = (epoch + week).toString()

        val resource = URLEncoder.encode(resourceUri, "UTF-8")
        val stringToSign = resource + "\n" + expiry
        val sigBytes = getHmaC256(key, stringToSign)
        val signature = URLEncoder.encode(
            sigBytes,
            "UTF-8"
        )
        return "SharedAccessSignature sr=" + resource + "&sig=" + signature + "&se=" + expiry + "&skn=" + keyName
    }

    fun getHmaC256(key: String, input: String): String {
        val sha256HMAC = Mac.getInstance("HmacSHA256")
        val secretKey = SecretKeySpec(key.toByteArray(), "HmacSHA256")
        sha256HMAC.init(secretKey)
        val encoder = Base64.getEncoder()
        return String(encoder.encode(sha256HMAC.doFinal(input.toByteArray())))
    }
}
